SessionPlayer.js

```diff
@@ -0,0 +1,174 @@
+                        },
+                        {
+                            stageDuration: [4, 2, 4, 0],
+                        },
+                    ]
+                },
+
+                { // Guides
+                    guideName: 'PowerPose',
+                    instructions: [
+                        { // An Instruction
+                            stageDuration: [15],
+                        },
+                        {
+                            stageDuration: [20],
+                        }
+                    ]
+                }
+            ];
+
+        this.state = {
+            guideState: 'playing', // 'playing', 'paused', 'finished'
+            currentGuideIndex: 0,
+            currentGuide: this.session[0].guide,
+            currentInstructionIndex: 0,
+            currentInstruction: this.session[0].instructions[0],
+        };
+    }
+
+    onInstructionEndHandler = () => {
+        incrementInstructionOrGuide();
+    }
+
+    incrementInstructionOrGuide = () => {
+        const nextInstruction =
+            this.session[currentGuideIndex]
+                .instructions[currentInstructionIndex + 1];
+
+        const nextGuide = this.session[currentGuideIndex + 1];
+
+        const {
+            currentInstructionIndex,
+            currentGuideIndex,
+            currentGuide,
+            currentInstruction
+        } = this.state;
+
+        if (typeof nextInstruction !== 'undefined') {
+            const nextInstructionIndex = currentInstructionIndex + 1;
+
+            this.setState({
+                currentInstructionIndex: nextInstructionIndex,
+                currentInstruction: currentGuide.instructions[nextInstructionIndex],
+            });
+        } else if (typeof nextGuide !== 'undefined') {
+            const nextGuideIndex = currentGuideIndex + 1;
+
+            this.setState({
+                currentInstructionIndex: 0,
+                currentInstruction: nextGuide.instructions[0],
+                currentGuideIndex: nextGuideIndex,
+                currentGuide: nextGuide,
+            });
+        } else {
+            console.log('All guides have been completed');
+        }
+    }
+
+    actionButtonHandler = () => {
+        const { guideState } = this.state;
+
+        switch(guideState) {
+            case 'playing':
+                this.setState({
+                    guideState: 'paused',
+                    currentInstructionIndex: 0,
+                    currentInstruction: this.session[0].instructions[0],
+                });
+                break;
+            case 'paused':
+                this.setState({ guideState: 'playing' });
+                break;
+            case 'finished':
+                incrementInstructionOrGuide();
+                break;
+        }
+    }
+
+    render() {
+        const {
+            currentGuide,
+            currentInstruction,
+            guideState,
+        } = this.state;
+
+        let DisplayedInstruction;
```

Comment:
I would pull render of `DisplayInstruction` into it's own function, but it is not crucial.

Reply:
I'm not that fond of breaking out functions for the sake of breaking out functions. This switch statement is core to the `render()` function, moving it elsewhere is just cargo-culting needless abstractions (especially since it's not used anywhere else).

Reply:
Sure